src/canvas/Canvas.ts

```diff
@@ -812,19 +812,20 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
     }
     let pointer, corner;
     if (target) {
-      corner = target._findTargetCorner(
+      const found = target._findTargetCorner(
         this.getViewportPoint(e),
         isTouchEvent(e)
       );
+      const { key, control } = found || {};
+      corner = key;
       if (
         target.selectable &&
         target !== this._activeObject &&
         target.activeOn === 'up'
       ) {
         this.setActiveObject(target, e);
         shouldRender = true;
-      } else {
-        const control = target.controls[corner];
+      } else if (control) {
```

Comment:
if you add `if control` here, the control check before the getMouseUpHandler is not necessary anymore.